### PR TITLE
Create task repository, added subtasks

### PR DESCRIPTION
**Pull request by hannachyzhyk**

### CODE REVIEW

File: src/database/schemas/task.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  status: {
+    required: true,
+    type: Number
+  },
+  createdAt: {
+    required: true,
+    type: Date
+  },
+  sortId: {
```
Comment (torgus81) [inline src/database/schemas/task.ts:17]: what's that for? Do you need extra prop for sorting?

File: src/database/schemas/task.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  deletedAt: {
+    required: false,
+    type: Date
+  },
+  writing: {
+    required: false,
+    type: String
+  },
+  parentId: {
+    required: false,
```
Comment (torgus81) [inline src/database/schemas/task.ts:34]: is it supposed for a task to have a linked structure like a tree or linked list?

File: src/repositories/repository.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export default interface IRepository<T> {
+    create(data: T): Promise<T>;
+    getAll(): Promise<T[]>;
+    get(id: string): Promise<T>;
+    update(id: string, data: any): Promise<T>;
```
Comment (torgus81) [inline src/repositories/repository.ts:5]: SUGGESTION:
to avoid `any` here, you could possibly add a separate model with nullable fields

**Review by torgus81 — APPROVED** [review]

File: src/database/schemas/task.ts (hunk 1 above)
Comment (hannachyzhyk) [reply src/database/schemas/task.ts:17]: Yes, if user whats to readjust list of task after creation I need to save this order

File: src/database/schemas/task.ts (hunk 2 above)
Comment (hannachyzhyk) [reply src/database/schemas/task.ts:34]: Child will have parentId this way I could build the tree of the tasks

File: src/repositories/repository.ts (hunk 3 above)
Comment (hannachyzhyk) [reply src/repositories/repository.ts:5]: Interesting idea, this interface will be used only in this case I suppose?

Comment (torgus81) [reply src/repositories/repository.ts:5]: yes, just a set of fields that possibly can get updated

File: src/database/schemas/task.ts (hunk 1 above)
Comment (torgus81) [reply src/database/schemas/task.ts:17]: is it `required` then? I guess it might be empty if the default sorting (using dates or any other prop) is used

**Pull request merged**